dask_cuda/cli/dask_cuda_worker.py

```diff
@@ -112,6 +112,15 @@
     "WARNING: managed memory is currently incompatible with NVLink, "
     "trying to enable both will result in an exception.",
 )
+@click.option(
+    "--rmm-log-directory",
+    default=None,
```

Comment:
When using a Pool and RMM is set to None will there be an error or will RMM not log ?

Reply:
Right now if `--disable-rmm-pool` is explicitly passed, there will be no logging. Do you have a preference on what should happen in that case @quasiben ?

Reply:
Is this referring to the `dask-cuda-worker` CLI or the benchmark options? In both cases, if `--rmm-log-directory` is set to None, then `logging` will be set to False in their respective RMM setups and it will be disabled.

Like @pentschev said, if `--disable-rmm-pool` is passed in the benchmarks the worker/scheduler's calls to `rmm.reintialize` will be skipped altogether and logging will be disabled. This relates somewhat to the conversation we had above - is there any utility to enabling RMM logging even if we aren't using a memory pool or managed memory?

Reply:
Is it possible to pass `--rmm-log-directory` without a directory? If so, what happens?

Reply:
Nice catch! If it's the last argument provided, it gives an error:

```
>>> dask-cuda-worker localhost:8786 --rmm-log-directory
Error: --rmm-log-directory option requires an argument
```

However, if it isn't the last argument, the following argument is interpreted as the directory to write to:

```
>>> dask-cuda-worker localhost:8786 --rmm-log-directory --no-dashboard
...
>>> ls -- --no-dashboard/
rmm_log_127.0.0.1:34325.dev0.txt  rmm_log_127.0.0.1:35957.dev0.txt 
```

Is there anything we can do to check for this?

Reply:
Yeah it would be good if it could error consistently. I'm sure there is a way for click to do this, but I don't know offhand what is

Reply:
The closest I could find is the [handling of option-like arguments](https://click.palletsprojects.com/en/7.x/arguments/#option-like-arguments), but I think that's limited to `click.arguments`; setting `ignore_unknown_options` to False in the command context didn't change anything here.

If we do find a general solution for this, it would probably be good to open up a separate PR applying it throughout the CLI, as it looks like most of the options behave in a similar fashion; they just usually lead to errors later on unrelated to click.

Reply:
> Like @pentschev said, if --disable-rmm-pool is passed in the benchmarks the worker/scheduler's calls to rmm.reintialize will be skipped altogether and logging will be disabled. This relates somewhat to the conversation we had above - is there any utility to enabling RMM logging even if we aren't using a memory pool or managed memory?

Somehow I thought this comment was for the benchmarks only. Regardless, I think @charlesbluca has already clarified.

> The closest I could find is the [handling of option-like arguments](https://click.palletsprojects.com/en/7.x/arguments/#option-like-arguments), but I think that's limited to `click.arguments`; setting `ignore_unknown_options` to False in the command context didn't change anything here.
> 
> If we do find a general solution for this, it would probably be good to open up a separate PR applying it throughout the CLI, as it looks like most of the options behave in a similar fashion; they just usually lead to errors later on unrelated to click.

Even though @jakirkham raises a good point, I wouldn't worry too much about it in this PR, as this is also an issue with all other arguments that require a value and somehow we didn't have too many issues with it. I think we could simply open an issue to eventually address that but not worry about it in this PR.